src/convolution/cnn.py

```diff
@@ -82,17 +129,16 @@ def main():
 
     channels_range = [config.filter_channels_start * (2 ** m) for m in
                       range(0, int(math.log(config.filter_channels_stop / config.filter_channels_start, 2) + 1))]
-
-    # descriptor?
-    for layer_cnt in range(config.layer_cnt_start, config.layer_cnt_stop + 1, config.layer_cnt_step):
-        for activation in config.activation_functions:
-            for channels in channels_range:
-                for loss_function in config.loss_functions:
-                    model = build_cnn(1, layer_cnt, activation, channels)
-                    model = train_model(model, gen_train, gen_validation, loss_function)
-
-                    evaluate_model(model, gen_test)
-                    keras.backend.clear_session()
+    # channels_range = [16]
+    for bottleneck_layers in config.bottleneck_layer_sizes:
+        for layer_cnt in range(config.layer_cnt_start, config.layer_cnt_stop + 1, config.layer_cnt_step):
+            for activation in config.activation_functions:
+                for channels in channels_range:
+                    for loss_function in config.loss_functions:
+                        model = build_cnn(1, layer_cnt, activation, channels, bottleneck_layers, bool(config.add_dropout))
+                        model = train_model(model, gen_train, gen_validation, loss_function)
+                        evaluate_model(model, gen_test)
+                        keras.backend.clear_session()
```

Comment:
sth lost in merges:
                    if config.save_cnn_model:
                       logging.info("Saving base_model")
                       model.save(config.base_model_path)